xclim/indices/_agro.py

```diff
@@ -341,3 +344,75 @@ def water_budget(
 
     out.attrs["units"] = pr.attrs["units"]
     return out
+
+
+@declare_units(pr="[precipitation]", thresh="[length]")
+def dry_spell_frequency(
+    pr: xarray.DataArray, thresh: str = "1.0 mm", window: int = 3, freq: str = "YS"
+) -> xarray.DataArray:
+    """
+    Return the number of dry periods of n days and more, during which the accumulated precipitation on a window of
+    n days is under the threshold.
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+      Daily precipitation.
+    thresh : str
+      Accumulated precipitation value under which a period is considered dry.
+    window : int
+      Number of days where the accumulated precipitation is under threshold.
+    freq : str
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+      The {freq} number of dry periods of minimum {window} days.
+    """
+    pram = rate2amount(pr, "mm")
+    pram.attrs["units"] = "mm"
+    thresh = convert_units_to(thresh, pram)
```

Comment:
Hum, not sure this will work? The 2nd argument to `rate2amout` is `dim='time'`, `out_units` is third.
Also, not sure I understood your answer to the other comment and this change : why force "mm"?
Why not:
```suggestion
    pram = rate2amount(pr)
    thresh = convert_units_to(thresh, pram)
```
?

Reply:
What I meant: I use rate2amount(pr) to convert pr values from kg/m2/s into mm. However, the attribute of pr reads like kg/m2 and not mm. So, thresh = convert_units_to(thresh, pram) returns an error, saying it cannot convert mm into kg/m2.

But I see now that if I specify the out_units like you said: pram = rate2amount(pr, out_units="mm"), pram units will be in mm. This way the thresh = convert_units_to(thresh, pram) will work.

Reply:
Oh! True, I didn't think of this first issue with kg/m²...